airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -111,7 +111,6 @@ def read(
                     )
 
                 try:
-                    self._apply_log_level_to_stream_logger(logger, stream_instance)
```

Comment:
that makes sense. The idea that any stream can define it's own logger isn't great and is conflicting with this idea that the source should be able to set the log level.

The current behavior is inconsistent since the log level of parent streams instantiated by the child won't updated.

That being said, I agree with you and I'll update the PR so
- the entrypoint sets the log level
- the source applies it's log level to the stream logger (this will keep the current inconsistent behavior)

Reply:
> The current behavior is inconsistent since the log level of parent streams instantiated by the child won't updated.

Actually, doing further testing, it seems like it applies. I was just misled it seems...

```
import logging
logging.basicConfig(format='%(asctime)s %(message)s')
airbyte_logger = logging.getLogger("airbyte")
airbyte_logger.setLevel(logging.DEBUG)

print(f"Parent logger level: {airbyte_logger.level}")
print(f"Child logger level: {logging.getLogger('airbyte.toto').level}")

airbyte_logger.warning("warning")
airbyte_logger.info("info")
airbyte_logger.debug("debug")

logging.getLogger("airbyte.toto").warning("child warning")
logging.getLogger("airbyte.toto").info("child info")
logging.getLogger("airbyte.toto").debug("child debug")
```

Output:
```
2023-11-15 09:32:24,062 warning
2023-11-15 09:32:24,062 info
2023-11-15 09:32:24,062 debug
2023-11-15 09:32:24,062 child warning
2023-11-15 09:32:24,062 child info
2023-11-15 09:32:24,062 child debug
Parent logger level: 10
Child logger level: 0
```

So even though they have different level, the logs are still being printed out to the console. Very weird that the level is different but it still works. I haven't fully understand why but eh! 

Hence, `logger.setLevel(logging.DEBUG)` should work. Do we need to source to apply the log level knowing this? Can we assume all loggers used by the source have `airbyte` as a parent?

Reply:
I don't feel too strongly about this. Assuming all loggers used by the source have `airbyte` as a parent is probably mostly true today, but it might not be true when embedding the CDK in other applications like native apps or embedding for LLMs. 

I'm fine keeping the the apply at the source level if it can be useful for these use cases